File: Models/DSSAT_ST_standalone/src/java/DSSAT_ST_standalone/STEMP_State.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;
public class STEMP_State
{
    private Double HDAY;
    private Double SRFTEMP;
    private Double [] ST;
    private Double [] TMA;
    private Double TDL;
    private Double CUMDPT;
    private Double ATOT;
    private Double [] DSMID;
    
    public STEMP_State() { }
    
    public STEMP_State(STEMP_State toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.HDAY = toCopy.getHDAY();
            this.SRFTEMP = toCopy.getSRFTEMP();
            ST = new Double[toCopy.getST().length];
        for (int i = 0; i < toCopy.getST().length; i++)
        {
            ST[i] = toCopy.getST()[i];
        }
            TMA = new Double[5];
        for (int i = 0; i < 5; i++)
        {
            TMA[i] = toCopy.getTMA()[i];
        }
            this.TDL = toCopy.getTDL();
            this.CUMDPT = toCopy.getCUMDPT();
            this.ATOT = toCopy.getATOT();
            DSMID = new Double[toCopy.getDSMID().length];
        for (int i = 0; i < toCopy.getDSMID().length; i++)
        {
            DSMID[i] = toCopy.getDSMID()[i];
        }
        }
    }
    public Double getHDAY()
    { return HDAY; }

    public void setHDAY(Double _HDAY)
    { this.HDAY= _HDAY; } 
    
    public Double getSRFTEMP()
    { return SRFTEMP; }

    public void setSRFTEMP(Double _SRFTEMP)
    { this.SRFTEMP= _SRFTEMP; } 
    
    public Double [] getST()
    { return ST; }

    public void setST(Double [] _ST)
    { this.ST= _ST; } 
    
    public Double [] getTMA()
    { return TMA; }

    public void setTMA(Double [] _TMA)
    { this.TMA= _TMA; } 
    
    public Double getTDL()
    { return TDL; }

    public void setTDL(Double _TDL)
    { this.TDL= _TDL; } 
    
    public Double getCUMDPT()
    { return CUMDPT; }

    public void setCUMDPT(Double _CUMDPT)
    { this.CUMDPT= _CUMDPT; } 
    
    public Double getATOT()
    { return ATOT; }

    public void setATOT(Double _ATOT)
    { this.ATOT= _ATOT; } 
    
    public Double [] getDSMID()
    { return DSMID; }

    public void setDSMID(Double [] _DSMID)
    { this.DSMID= _DSMID; } 
    
}

File: Models/DSSAT_ST_standalone/src/cpp/DSSAT_ST_standalone/STEMP_State.cpp
#include "STEMP_State.h"
using namespace DSSAT_ST_standalone;

STEMP_State::STEMP_State() {}

double STEMP_State::getHDAY() { return this->HDAY; }
double STEMP_State::getSRFTEMP() { return this->SRFTEMP; }
std::vector<double> & STEMP_State::getST() { return this->ST; }
std::vector<double> & STEMP_State::getTMA() { return this->TMA; }
double STEMP_State::getTDL() { return this->TDL; }
double STEMP_State::getCUMDPT() { return this->CUMDPT; }
double STEMP_State::getATOT() { return this->ATOT; }
std::vector<double> & STEMP_State::getDSMID() { return this->DSMID; }

void STEMP_State::setHDAY(double _HDAY) { this->HDAY = _HDAY; }
void STEMP_State::setSRFTEMP(double _SRFTEMP) { this->SRFTEMP = _SRFTEMP; }
void STEMP_State::setST(std::vector<double> const &_ST){
    this->ST = _ST;
}
void STEMP_State::setTMA(std::vector<double> const &_TMA){
    this->TMA = _TMA;
}
void STEMP_State::setTDL(double _TDL) { this->TDL = _TDL; }
void STEMP_State::setCUMDPT(double _CUMDPT) { this->CUMDPT = _CUMDPT; }
void STEMP_State::setATOT(double _ATOT) { this->ATOT = _ATOT; }
void STEMP_State::setDSMID(std::vector<double> const &_DSMID){
    this->DSMID = _DSMID;
}